ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudConfig.java

```diff
@@ -71,6 +83,119 @@
       "com.github.ambry.cloud.azure.ConnectionStringBasedStorageClient";
   public static final String USE_ASYNC_AZURE_APIS = "use.async.azure.apis";
 
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_STR = "storageAccountInfo";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_NAME = "name";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_PARTITION_RANGE = "partitionRange";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_STORAGE_SCOPE = "storageScope";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_STORAGE_ENDPOINT = "storageEndpoint";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_STORAGE_CONNECTION_STRING = "storageConnectionString";
+
+  /**
+   * Stores Azure Storage Account related info for one storage account.
+   */
+  public static class StorageAccountInfo {
+    private final String name;
+    private final int partitionRangeStart;
+    private final int partitionRangeEnd;
+    private final String storageScope;
+    private final String storageEndpoint;
+    private final String storageConnectionString;
+
+    /**
+     * Construct a StorageAccountInfo object with the given parameters.
+     * @param name the name of the storage account.
+     * @param partitionRangeStart the lower bound (inclusive) of the partition range this storage account covers.
+     * @param partitionRangeEnd the upper bound (exclusive) of the partition range this storage account covers.
+     * @param storageScope the Azure scope.
+     * @param storageEndpoint the Azure end-point
+     */
+    StorageAccountInfo(String name, int partitionRangeStart, int partitionRangeEnd, String storageScope, String storageEndpoint, String storageConnectionString) {
+      this.name = name;
+      this.partitionRangeStart = partitionRangeStart;
+      this.partitionRangeEnd = partitionRangeEnd;
+
+      // Used by ADAuthBasedStorageClient
+      this.storageScope = storageScope;
+
+      // Used by ADAuthBasedStorageClient and ClientSecretCredentialStorageClient
+      this.storageEndpoint = storageEndpoint;
+
+      // Used by ConnectionStringBasedStorageClient
+      this.storageConnectionString = storageConnectionString;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public int getPartitionRangeStart() {
+      return partitionRangeStart;
+    }
+
+    public int getPartitionRangeEnd() {
+      return partitionRangeEnd;
+    }
+
+    public String getStorageScope() {
+      return storageScope;
+    }
+
+    public String getStorageEndpoint() {
+      return storageEndpoint;
+    }
+
+    public String getStorageConnectionString() { return storageConnectionString; }
+  }
+
+  static List<StorageAccountInfo> parseStorageAccountInfo(AzureCloudConfig azureCloudConfig) {
```

Comment:
missing java docs in this and other methods of this class.

Reply:
Done